drivers/ethernet/eth_numaker.c

```diff
@@ -0,0 +1,807 @@
+
+	if (offload_needed) {
+		/*
+		 * Make sure that the OS you are running supports the IP and TCP checksum
+		 * offloading, before calling any of the functions given below.
+		 */
+		synopGMAC_tx_checksum_offload_tcp_pseudo(gmacdev, txdesc);
+	} else {
+		synopGMAC_tx_checksum_offload_bypass(gmacdev, txdesc);
+	}
+	__DSB();
+	txdesc->status |= DescOwnByDma;
+
+	gmacdev->TxNext = synopGMAC_is_last_tx_desc(gmacdev, txdesc) ? 0 : txnext + 1;
+	gmacdev->TxNextDesc =
+		synopGMAC_is_last_tx_desc(gmacdev, txdesc) ? gmacdev->TxDesc : (txdesc + 1);
+
+	/* Enable the interrupt */
+	synopGMAC_enable_interrupt(gmacdev, DmaIntEnable);
+	/* Trigger TX DMA */
+	synopGMAC_resume_dma_tx(gmacdev);
+}
+
+static int numaker_eth_tx(const struct device *dev, struct net_pkt *pkt)
+{
+	struct eth_numaker_data *data = dev->data;
+	synopGMACdevice *gmacdev = data->gmacdev;
+	uint16_t total_len = net_pkt_get_len(pkt);
+	uint8_t *buffer = NULL;
+
+	/* Get exclusive access */
+	k_mutex_lock(&data->tx_frame_buf_mutex, K_FOREVER);
+
+	if (total_len > NET_ETH_MAX_FRAME_SIZE) {
+		/* NuMaker SDK reserve 2048 for tx_buf */
+		LOG_ERR("TX packet length [%d] over max [%d]", total_len, NET_ETH_MAX_FRAME_SIZE);
+		goto error;
+	}
+
+	buffer = m_numaker_gmacdev_get_tx_buf(gmacdev);
+	LOG_DBG("buffer=0x%x", (uint32_t)buffer);
+	if (buffer == NULL) {
+		goto error;
+	}
+
+	if (net_pkt_read(pkt, buffer, total_len)) {
+		goto error;
+	}
+
+	/* Prepare transmit descriptors to give to DMA */
+	m_numaker_gmacdev_trigger_tx(gmacdev, total_len);
+
+	k_mutex_unlock(&data->tx_frame_buf_mutex);
+
+	return 0;
+
+error:
+	LOG_ERR("Writing pkt to TX descriptor failed");
+	k_mutex_unlock(&data->tx_frame_buf_mutex);
+	return -EIO;
+}
+
+static void numaker_eth_if_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct eth_numaker_data *data = dev->data;
+
+	synopGMACdevice *gmacdev = data->gmacdev;
+
+	LOG_DBG("eth_if_init");
+
+	/* Read mac address */
+	m_numaker_read_mac_addr(data->mac_addr);
+
+	net_if_set_link_addr(iface, data->mac_addr, sizeof(data->mac_addr), NET_LINK_ETHERNET);
+	data->iface = iface;
+	ethernet_init(iface);
+
+	/* Enable GMAC device INT & TX/RX */
+	m_numaker_gmacdev_enable(gmacdev);
+}
+
+static int numaker_eth_set_config(const struct device *dev, enum ethernet_config_type type,
+				  const struct ethernet_config *config)
+{
+	struct eth_numaker_data *data = dev->data;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
+		memcpy(data->mac_addr, config->mac_address.addr, sizeof(data->mac_addr));
+		synopGMAC_set_mac_address(NUMAKER_GMAC_INTF, data->mac_addr);
+		net_if_set_link_addr(data->iface, data->mac_addr, sizeof(data->mac_addr),
+				     NET_LINK_ETHERNET);
+		LOG_DBG("%s MAC set to %02x:%02x:%02x:%02x:%02x:%02x", dev->name, data->mac_addr[0],
+			data->mac_addr[1], data->mac_addr[2], data->mac_addr[3], data->mac_addr[4],
+			data->mac_addr[5]);
+		return 0;
+	default:
+		break;
```

Comment:
nit: I would `return -ENOTSUP` here as well, then I think you could even drop the return in line "557", the compiler should be happy with it